src/store/ChartState.js

```diff
@@ -80,6 +80,11 @@ class ChartState {
             this.stxx.isAutoScale = this.settings && settings.isAutoScale !== false;
         }
         if (this.granularity !== granularity && this.context) {
+            if (calculateTimeUnitInterval(granularity).timeUnit === 'second' && this.mainStore.chartType.isCandle) {
```

Comment:
@nazaninreihani Also, you should check the value of `chartType`. What happens if the value of `this.mainStore.chartType.isCandle` is not `true` but the value of `chartType` is a candle type?